### DISCUSSION TITLE
Assets key for Frontmatter CMS

### DISCUSSION THREAD

**Discussion by louiss0** [estruyf/vscode-front-matter#542] [Ideas]
I have decided to talk about this as a feature idea because. I didn't think of the problem as an issue as in bug. But what I'm going to talk about is a restriction.  As of today you are only allowed to use one assets folder. This is not good for people like me who use mono-repos. 

![frontmatter-public-folder-snip](https://user-images.githubusercontent.com/29028601/226142951-4a84e74f-0d7e-41f4-b1ef-e995a23b19c0.png)

Right now this is all I can do. If can could specify multiple files they would look like this.

![frontmatter-asset-folders-snip](https://user-images.githubusercontent.com/29028601/226143002-43da6bfc-483b-41bc-9687-39cba96fbc77.png)

This is what I want to do. The benefit of this is that the key could be the name of the place where each asset is displayed. The value is the folder to use. This means that pictures and where they come from can be specified in a mono-repo. even outside of a mono-repo developers can organize where each set of images are in a folder.

**Comment by estruyf** [comment]
Thank you for opening this discussion @louiss0. Think this would indeed be great for people that split their assets from the website.

What is the URL of the images you are inserting in your articles?

**Reply by louiss0** [reply]
I'm not sure how to answer this question but I think It's "frontMatter.content.publicFolder": "[[workspace]]/apps/blog/public",

**Reply by estruyf** [reply]
@louiss0 I want to know how your images are referenced in your content. What is the path to the pictures of the content?

Like:

```markdown
![](./link/to/the/image.png)
```

I want to understand better if there is a difference between the blog and the portfolio.

**Reply by louiss0** [reply]
> I'm not sure how to answer this question but I think It's "frontMatter.content.publicFolder": "[[workspace]]/apps/blog/public",

The blog folder is the folder that I use for images for my blog. The portfolio is the one for my portfolio.  I reference my images using `/assets/image.png`. With Astro, I'm stuck with the public folder for each for now. I can't optimize images when using Astro in non-Astro Components.

**Reply by estruyf** [reply]
Does it mean you have two types of content placed in two individual projects/sites?

If you use `/assets/image.png`, is the image loaded because of the path of your `/blog` or `/portfolio`?

Another way would be to provide an assets path on the content type.

**Reply by louiss0** [reply]
> > I'm not sure how to answer this question but I think It's "frontMatter.content.publicFolder": "[[workspace]]/apps/blog/public",
> 
> The blog folder is the folder that I use for images for my blog. The portfolio is the one for my portfolio. I reference my images using `/assets/image.png`. With Astro, I'm stuck with the public folder for each for now. I can't optimize images when using Astro in non-Astro Components.

I have two folders for different images. I'd rather do that. I have to do that so that Astro could bundle them. In Astro I have to create a `/public` folder for the blog and the portfolio folders. Soon in Astro I'll have to create an assets folder in each project for them to be bundled.

**Reply by estruyf** [reply]
I get you to have two different image folders in two project folders (blog and portfolio). What I am still missing is, do you have two Astro sites. One for **blog** and one for **portfolio**?

Might be helpful to show a screenshot of your monorepo tree view.

**Reply by louiss0** [reply]
> I get you to have two different image folders in two project folders (blog and portfolio). What I am still missing is, do you have two Astro sites. One for **blog** and one for **portfolio**?
> 
> Might be helpful to show a screenshot of your monorepo tree view.

**Here is a snapshot of my repo**

![Screenshot 2023-03-23 110356](https://user-images.githubusercontent.com/29028601/227246093-f13bc549-ea4f-48e4-b046-3aad10bf5374.png)

As you should see the apps folder is where all of my apps are. The blog and the portfolio folder live there.

**Reply by louiss0** [reply]
It seems like you don't understand what is going on. I have an Nx mono-repo where I use the a folder called the `apps` folder to hold my projects. So far both of them are `blog` and `portfolio`. The apps folder lives in the root. The other folders don't matter in this conversation. I plan on storing the pictures for my blog posts inside of the blog folder. The ones for my `portfolio` will live in the portfolio folder. 

The way Astro woks is that I need to have a `/public` folder for unbundled assets. It will not process them at all. I think there is a way to override this but I won't do it. Even if I could do that I'd have to tell Astro to bundle different assets for different projects. Sooner or later Astro will add another special directory called  `assets` this one has to live in the `src` directory and must be referenced so that images are bundled. If you create images from directory then use them the images will be built and optimized.

**Reply by estruyf** [reply]
> I get you to have two different image folders in two project folders (blog and portfolio). What I am still missing is, do you have two Astro sites. One for blog and one for portfolio?

So the `blog` and `portfolio` are two separate websites. I got it now.

As it was not evident initially, I needed to understand this. It could have been that you split up these images into two folders as you wanted to push these to an image service like Cloudinary.

Will have to think on how Front Matter will be able to support this in a convenient way. Your suggestion is a good one; also still thinking about bringing it to the page folder or content type level, as that is closer to the content configuration itself. It might be that the page folder makes more sense.

**Reply by estruyf** [reply]
Created an issue to work on it: https://github.com/estruyf/vscode-front-matter/issues/548
